motor/core.py

```diff
@@ -146,7 +147,12 @@ def __init__(self, *args, **kwargs):
 
         delegate = self.__delegate_class__(*args, **kwargs)
         super().__init__(delegate)
-        self.io_loop = io_loop
+
+    @property
+    def io_loop(self):
+        if self._io_loop is None:
+            self._io_loop = self._framework.get_event_loop()
```

Comment:
we might need to make a similar change in AgnosticClientEncryption.__init__.

Reply:
Done